components/RouteTestDialog.tsx

```diff
@@ -0,0 +1,1049 @@
+  setTokenType: (v: "user" | "bot" | "bearer" | null) => void;
+  body: string;
+  setBody: (v: string) => void;
+  customHeaders: { key: string; value: string }[];
+  setCustomHeaders: React.Dispatch<React.SetStateAction<{ key: string; value: string }[]>>;
+  activeRequestTab: "path" | "query" | "headers" | "body";
+  setActiveRequestTab: (v: "path" | "query" | "headers" | "body") => void;
+  response: {
+    status: number;
+    statusText: string;
+    headers: Record<string, string>;
+    body: unknown;
+  } | null;
+  activeTab: "body" | "headers";
+  setActiveTab: (v: "body" | "headers") => void;
+  handleKeyDown: (e: React.KeyboardEvent) => void;
+}
+
+function RequestView({
+  method,
+  url,
+  pathParams,
+  setPathParams,
+  queryParams,
+  setQueryParams,
+  optionalQueryParams,
+  token,
+  setToken,
+  tokenType,
+  setTokenType,
+  body,
+  setBody,
+  customHeaders,
+  setCustomHeaders,
+  activeRequestTab,
+  setActiveRequestTab,
+  response,
+  activeTab,
+  setActiveTab,
+  handleKeyDown,
+}: RequestViewProps) {
+  return (
+    <>
+      <div className="mb-4 flex items-center gap-2">
+        <MethodBadge method={method} />
+        <code className="break-all text-base text-text-light dark:text-text-dark">
+          {Object.entries(pathParams).reduce((acc, [key, value]) => {
+            return value ? acc.replace(`{${key}}`, value) : acc;
+          }, url)}
+        </code>
+      </div>
+      <div className="flex flex-col gap-4 pb-4">
+        <div>
+          <label className={Styles.dialogLabel}>
+            Authorization Token
+            {tokenType === "user" && (
+              <>
+                <span className="ml-2 font-normal">—</span>
+                <span className="ml-2 font-normal text-red-500">
+                  Automating user accounts is against platform Terms of Service. Proceed with caution.
+                </span>
+              </>
+            )}
+            {tokenType === "bot" && (
+              <>
+                <span className="ml-2 font-normal">—</span>
+                <span className="ml-2 font-normal text-orange-500">Bot tokens are blocked in browsers.</span>
+              </>
+            )}
+          </label>
+          <input
+            type="password"
+            className={classNames(Styles.dialogInput, {
+              "!border-red-500 focus:!border-red-500 focus:!ring-red-500": tokenType == null && token.length > 0,
+            })}
+            placeholder="User or bearer token"
+            value={token}
+            onKeyDown={handleKeyDown}
+            onChange={(e) => {
+              const val = e.target.value;
+              setToken(val);
+              localStorage.setItem("discord_api_token", val);
+
+              if (BOT_TOKEN_REGEX.test(val)) setTokenType("bot");
+              else if (USER_TOKEN_REGEX.test(val)) setTokenType("user");
+              else if (BEARER_TOKEN_REGEX.test(val)) setTokenType("bearer");
```

Comment:
Is there a need for complex regexes on every change? Maybe simplify to .startsWith or run on focus change to another form element